[inductor] Update triton pin

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99661
* #99660
* #99484
* __->__ #99209

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

peterbell10 committed:
[TESTING] Update triton pin

`triton.testing.do_bench` on master has a new signature with different
default behaviour. This updates our users and adds a small wrapper
`run_benchmark` so we can uniformly decide what metric to use. Here
I've kept median for BC, while triton has changed to mean.

[ghstack-poisoned]

ngimel commented:
#99216 adapts inductor do_bench to triton changes.

ngimel commented:
Previous pin update fails due to https://github.com/pytorch/pytorch/pull/98728#issuecomment-1509467765

peterbell10 committed:
Update on "[TESTING] Update triton pin"


`triton.testing.do_bench` on master has a new signature with different
default behaviour. This updates our users and adds a small wrapper
`run_benchmark` so we can uniformly decide what metric to use. Here
I've kept median for BC, while triton has changed to mean.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 committed:
Update on "[TESTING] Update triton pin"


`triton.testing.do_bench` on master has a new signature with different
default behaviour. This updates our users and adds a small wrapper
`run_benchmark` so we can uniformly decide what metric to use. Here
I've kept median for BC, while triton has changed to mean.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 committed:
Update on "[TESTING] Update triton pin"


`triton.testing.do_bench` on master has a new signature with different
default behaviour. This updates our users and adds a small wrapper
`run_benchmark` so we can uniformly decide what metric to use. Here
I've kept median for BC, while triton has changed to mean.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 committed:
Update on "[TESTING] Update triton pin"


`triton.testing.do_bench` on master has a new signature with different
default behaviour. This updates our users and adds a small wrapper
`run_benchmark` so we can uniformly decide what metric to use. Here
I've kept median for BC, while triton has changed to mean.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 committed:
Rebase on "[TESTING] Update triton pin"


`triton.testing.do_bench` on master has a new signature with different
default behaviour. This updates our users and adds a small wrapper
`run_benchmark` so we can uniformly decide what metric to use. Here
I've kept median for BC, while triton has changed to mean.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 committed:
Update on "[TESTING] Update triton pin"


`triton.testing.do_bench` on master has a new signature with different
default behaviour. This updates our users and adds a small wrapper
`run_benchmark` so we can uniformly decide what metric to use. Here
I've kept median for BC, while triton has changed to mean.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge

peterbell10 commented:
@pytorchbot merge -f "dynamo failure is pre-existing"
